### ISSUE TITLE
Plotting of shared models (parallely reused models)

### ISSUE THREAD

**Issue by johschmitz** [keras-team/keras#11492]
Consider the following example  and the resulting `model.png`
(Basically a slightly modified version of https://keras.io/getting-started/functional-api-guide/#shared_vision_model)

    from keras.layers import Conv2D, MaxPooling2D, Input, Dense, Flatten, concatenate
    from keras.models import Model
    from keras.utils import plot_model

    # First, define the vision modules
    digit_input = Input(shape=(27, 27, 1))
    x = Conv2D(64, (3, 3))(digit_input)
    x = Conv2D(64, (3, 3))(x)
    x = MaxPooling2D((2, 2))(x)
    out = Flatten()(x)

    vision_model = Model(digit_input, out)

    # Then define the tell-digits-apart model
    digit_a = Input(shape=(27, 27, 1))
    digit_b = Input(shape=(27, 27, 1))

    # The vision model will be shared, weights and all
    out_a = vision_model(digit_a)
    out_b = vision_model(digit_b)

    concatenated = concatenate([out_a, out_b])
    out = Dense(1, activation='sigmoid')(concatenated)

    classification_model = Model([digit_a, digit_b], out)

    plot_model(classification_model, show_shapes=True, to_file='model.png')

![model](https://user-images.githubusercontent.com/771979/47510895-5dbb5280-d879-11e8-9207-b7733b6b708f.png)

How can the problem be solved that the "Model"/vision_model is shown only once in the plot and not twice in parallel?

**Comment by alexpyattaev** [comment]
There does not appear to be a solution, but it is very annoying issue, especially for more complex networks

**Comment by NoamGaash** [comment]
minimal reproducible  code:
```
from tensorflow.keras.layers import Dense, Input
from tensorflow.keras.models import Model
from tensorflow.keras.utils import plot_model

layer1 = Dense(50)
layer2 = Dense(40)

inp = Input((40, ))
x = layer1(layer2(layer1(inp)))
plot_model(Model(inp, x))
```

result:

![image](https://user-images.githubusercontent.com/11145132/70231194-868f6600-1762-11ea-98f9-17e14027503a.png)
